controllers/front/ProductController.php

```diff
@@ -476,6 +476,9 @@ protected function assignAttributesGroups($product_for_template = null)
                                 $current_cover = $image;
                             }
                         }
+                        if (!isset($current_cover)) {
+                            $current_cover = array_values($this->context->smarty->tpl_vars['product']->value['images'])[0];
```

Comment:
If not, a lot of serious problems will happens (and far before this call, believe me)